### DISCUSSION TITLE
Reinitialising an existing users database

### DISCUSSION THREAD

**Discussion by needledrag** [Mimetis/Dotmim.Sync#585] [Q&A]
Hey all, 

I've got a scope that contains tables that have a SyncDirection of UploadOnly.  Basicially user data that is input and used in the app.
If the user gets a new device (ie they already have an account with data in the server database) and reinstalls the app I call a Sync with a SyncType of Reinitialize with the idea that all the data they have input and was uploaded to the server can be reinstated on their device.  Unfortunately when I use Reinitialize with tables that have a SyncDirection of UploadOnly no data is brought down to the client.  In  a way it makes sense since the SyncDirection of the all the tables is set to UploadOnly but I was wondering if this is by design?    Is there another way for the client to tell the server it wants to Reinitialize if the tables are set to UploadOnly?  
If I use SyncDirection.BiDirectional it works perfect with the reinitialize but since the data that is input from the client is never changed except by the client I was hoping to have as little overhead as possible and only have the tables as UploadOnly.

Another thing I came across, if I have a table of of SyncDirection.DownloadOnly (a different scope) and use Columns.AddRange to limit some of the columns coming down, I automatically getting a tracking table created on the client for that table?  Is this also by design?  I'm just trying to understand a little more as to why it needs a tracking table on the client for data that will only ever be downloaded to the client?

Any suggestions greatly appreciated.

Regards
Shaun

**Comment by Mimetis** [comment]
> In a way it makes sense since the SyncDirection of the all the tables is set to UploadOnly but I was wondering if this is by design? Is there another way for the client to tell the server it wants to Reinitialize if the tables are set to UploadOnly?

Indeed, it's by design
I guess in your situation, an easy solution would be to use multi scopes;:
- One scope with tables with `UploadOnly`
- One scope with tables with `Bidirectional` mode, where you are doing the `Reinitiliaze` 

I did not tested this solution, but I guess it could work.
Let me know

> if I have a table of of SyncDirection.DownloadOnly , I automatically getting a tracking table created on the client for that table? 

It's a good question :)
Today, all tables that are part of the sync process, have tracking tables associated.
But what you are thinking makes perfect sense, theoretically, having a `SyncDirection.DownloadOnly` on table should not required a tracking  table. 
It's not the case today, but definitively something we can think of, in a future version

**Reply by needledrag** [reply]
Thanks for the reply, it works fine with Bidirectional so I'll do that, even if there is no insert/updates on the server,. Just trying to shave off as much processing as possible so the tracking tables aren't queried.  Overhead should be minimal though.  Thank you
